ceph_salt/config_shell.py

```diff
@@ -1227,8 +1230,27 @@ def generate_config_shell_tree(shell):
 class CephSaltConfigShell(configshell.ConfigShell):
     # pylint: disable=anomalous-backslash-in-string
     def __init__(self):
-        super(CephSaltConfigShell, self).__init__(
-            '~/.ceph_salt_config_shell')
+        #
+        # before initializing configshell_fb, check for zero-length prefs.bin
+        configshell_path = '{}/.ceph_salt_config_shell'.format(
+            os.path.expanduser('~'))
+        prefs_bin_path = '{}/prefs.bin'.format(configshell_path)
+        if os.path.exists(prefs_bin_path):
+            logger.debug('{} exists'.format(prefs_bin_path))
+            prefs_bin_filesize = os.path.getsize(prefs_bin_path)
+            if prefs_bin_filesize == 0:
+                logger.warning('{} has size 0; removing!'.format(prefs_bin_path))
+                os.remove(prefs_bin_path)
+        else:
+            logger.debug('{} does not exist'.format(prefs_bin_path))
+        #   
+        # initialize configshell_fb 
+        try:
+            super(CephSaltConfigShell, self).__init__(configshell_path)
+        except Exception as exc:
+            logger.critical(traceback.format_exc())
```

Comment:
```suggestion
            logger.exception("configshell_fb failed to initialize")
```

`logger.exception` takes care of writing the traceback

Reply:
ok, done (and tested - works fine)